### PR TITLE
[fix] receive/ and receive-out-of-band/ sometimes return HTTP 204 e…

### PR DESCRIPTION
**Pull request by rwrzesien**
…ven if there's a message queued

Resolves https://github.com/golemfactory/concent/issues/44
https://www.pivotaltracker.com/story/show/154401776

### CODE REVIEW

File: concent_api/core/tests/test_core_views.py [hunk 1]
```diff
@@ -379,6 +379,8 @@ def setUp(self):
         self.force_golem_data.task_to_compute = self.task_to_compute
         assert self.force_golem_data.timestamp == 1510912800
 
+    @freeze_time("2017-11-17 11:40:00")
+    def test_view_receive_out_of_band_should_accept_valid_message(self):
```
Comment (cameel) [inline concent_api/core/tests/test_core_views.py]: Where does this come from? Was the test broken? Please rebase the branch on master.

If you're fixing something unrelated to the issue then this should be a separate commit that explains what and why you're fixing.

Comment (rwrzesien) [reply concent_api/core/tests/test_core_views.py]: Please check the whole class, which have two tests of which one was broken, but it is related to what I am doing in current issue. There was message creation in setUp(), and second test name is `test_view_receive_out_of_band_return_http_204_if_no_messages_in_database`, which was wrong because test was passing and there was a message in database, so I have moved creation of it from setup to a first test which only needed it.

**Comment by rwrzesien** [comment]
@cameel Updated.

**Comment by rwrzesien** [comment]
@cameel Rebased on master, please also consider this PR as little refactoring.

**Comment by rwrzesien** [comment]
@cameel I have squashed fixups, there were no conflicts.

**Comment by rwrzesien** [comment]
Closing as discussed, general issue related with this case will be covered in different implementation.
